### PR TITLE
Upgrade containerd to 1.3.9

### PR DESCRIPTION
**Pull request by kzys**
- The version has a CVE fix, while we may not be affected.
- It is better to use relesed versions if possible.

Signed-off-by: Kazuyoshi Kato <katokazu@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: firecracker-control/local.go [hunk 1]
```diff
@@ -223,14 +223,14 @@ func (s *local) shimFirecrackerClient(requestCtx context.Context, vmID string) (
...
-	socketAddr, err := fcShim.FCControlSocketAddress(requestCtx, vmID)
+	socketAddr, err := fcShim.FCControlSocketAddress(requestCtx, s.containerdAddress, vmID)
 	if err != nil {
 		err = errors.Wrap(err, "failed to get shim's fccontrol socket address")
 		s.logger.WithError(err).Error()
 		return nil, err
 	}
 
-	return fcclient.New("\x00" + socketAddr)
+	return fcclient.New(socketAddr) //
```
Comment (samuelkarp) [inline firecracker-control/local.go:233]: Looks like a stray `//` got left in.

**Review by samuelkarp — APPROVED** [review]
LGTM! Just to confirm: CVE-2020-15257 does not affect firecracker-containerd as the abstract sockets exist outside the microVM guest; a process inside the microVM guest regardless of what namespaces and privileges it has cannot access an abstract socket on the host.

File: runtime/service.go [hunk 2]
```diff
@@ -354,6 +355,15 @@ func (s *service) StartShim(shimCtx context.Context, containerID, containerdBina
...
 	}
 
+	// If there is a live socket, reuse the socket without calling CreateVM.
+	address, err := shim.SocketAddress(shimCtx, containerdAddress, s.vmID)
+	if err != nil {
+		return "", err
+	}
+	if shim.CanConnect(address) {
+		return address, nil
+	}
```
Comment (kzys) [inline runtime/service.go:365]: I would double-check this part. While this code itself makes sense, I don't know why it *was* working when the sockets were abstract.

Comment (kzys) [reply runtime/service.go]: This change is not needed. So I've removed from the last revision.

**Comment by kzys** [comment]
We may skip this version and use 1.4.x. See #467.

**Pull request closed without merging**